cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -156,6 +156,17 @@ void set_error_for_invalid_utf(int64_t execution_context, char val) {
   free(error);
 }
 
+gdv_int32 validate_utf8_following_bytes(const char* data, gdv_int32 data_len,
+                                        gdv_int32 char_index, char* invalid_char) {
+  for (int j = 1; j < data_len; ++j) {
+    if ((data[char_index + j] & 0xC0) != 0x80) {  // bytes following head-byte of glyph
+      *invalid_char = data[char_index + j];
```

Comment:
why is this invalid_char (i think you meant invalid_byte) is needed? You are doing a unnecessary allocation for this.